src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -117,6 +202,22 @@ public Uri GetEndpointForClient(ConfigurationClient client)
             ConfigurationClientWrapper currentClient = _clients.FirstOrDefault(c => c.Client == client);
             
             return currentClient?.Endpoint;
+        }
+
+        private bool IsEligibleToRemove(IEnumerable<string> srvEndpointHosts, ConfigurationClientWrapper client)
+        {
+            // Only remove the client if it is auto failover, as well as not in returned SRV records.

```

Comment:
I don't see why. If they enable replica discovery then I think our srv lookups should be a source of truth.

Reply:
if users provide an explicit list of replica endpoints, shouldn't those be given preference for fetching config? I feel like srv replicas should be used when configured replicas are not available.

Reply:
Preference perhaps, but this is removing the replica because we didn't discover it in srv records. The scenario where it matters if someone lists a replica that is deleted during runtime. Srv record will no longer contain that replica. So do we trust the customers list that it should exist, or do we trust our infrastructure saying that replica does not exist.

Reply:
I expect the same as @avanigupta said. The user-provided endpoint(s) should have higher priority than those auto-discovered.

BTW, the "auto failover" comment should be updated.

Reply:
@zhenlan not sure if you saw my response before posting. Did you?

Reply:
@jimmyca15 thanks, no, I didn't :)

I think we should stick to the user-provided list. It could be a replica deleted accidentally or temporarily for some reason. It could be a replica to be created.

Reply:
* Replica deleted (accidentally or temporarily)
* Replica to be created

In all those cases, we shouldn't call that replica. And if it's temporary, a future query of srv record will resolve the issue.

The concern I would say would justify always including the user-provided replicas is if we don't trust our srv records to be accurate.

Reply:
If the customer do replication themself, create two stores that always have the same settings. We should not remove any user's clients from list.

Reply:
> If the customer do replication themself, create two stores that always have the same settings. We should not remove any user's clients from list.

I don't consider that a supported scenario. If we could, I might suggest to block it. 

However, I do agree that we 

1. Try all provided clients first
2. Try fallback clients afterward